packages/mattermost-redux/src/reducers/entities/emojis.ts

```diff
@@ -11,22 +12,27 @@ import {IDMappedObjects} from 'mattermost-redux/types/utilities';
 export function customEmoji(state: IDMappedObjects<CustomEmoji> = {}, action: GenericAction): IDMappedObjects<CustomEmoji> {
     switch (action.type) {
     case EmojiTypes.RECEIVED_CUSTOM_EMOJI: {
-        const nextState = {...state};
-        nextState[action.data.id] = action.data;
-        return nextState;
+        const emoji: CustomEmoji = action.data;
```

Comment:
why the extra variable init and no immediate return?

Reply:
I started doing that recently so that we get some clearer typing. Since our reducers aren't very TS-friendly, I'm using it like inline documentation about what data the action includes.

Reply:
Fair enough! 👍🏻